src/main/java/com/github/mauricioaniche/ck/CKVisitor.java

```diff
@@ -56,21 +56,22 @@ public CKVisitor(String sourceFilePath, CompilationUnit cu, Callable<List<ClassL
 	public boolean visit(TypeDeclaration node) {
 		ITypeBinding binding = node.resolveBinding();
 
-		// there might be metrics that use it
-		// (even before a class is declared)
-		if(!classes.isEmpty()) {
-			classes.peek().classLevelMetrics.stream().map(metric -> (CKASTVisitor) metric).forEach(ast -> ast.visit(node));
-			if (!classes.peek().methods.isEmpty())
-				classes.peek().methods.peek().methodLevelMetrics.stream().map(metric -> (CKASTVisitor) metric).forEach(ast -> ast.visit(node));
-		}
-
 		// build a CKClassResult based on the current type
 		// declaration we are visiting
 		String className = binding != null ? binding.getBinaryName() : node.getName().getFullyQualifiedName();
 		String type = getTypeOfTheUnit(node);
 		int modifiers = node.getModifiers();
 		CKClassResult currentClass = new CKClassResult(sourceFilePath, className, type, modifiers);
 		currentClass.setLoc(calculate(node.toString()));
+		
```

Comment:
Why was this moved here?

I guess the idea of having it before creating the new node is that we notify the current class about a `TypeDeclaration` happening inside of it. Then, we stack a new `CKClassResult`.

Reply:
I moved this part to get the current class name and set it for the AST, case the code inside the if block is called. If I don't get the current class name before, the program may fail.

Reply:
This is interesting, because it should not make a difference at all. The _CKClassResult_ of the class that was just instantiated is _not_ yet in the stack of classes. It only goes to the stack in now line 77, right? 

So, the `classes.peek().classLevelMetrics.stream().map` that happens inside the `if` you moved, will not go through the newly declared type. 

So, in a way, this code block being here has the same behaviour as the block where it was. What am I missing here?

Reply:
Yes! You're right, @mauricioaniche. The code block on line 77 now has the same behavior as it did before I moved it. The only reason I changed it was because I need to pass the class name as a parameter to ast now. So this change allows me to take the class name and pass it as a parameter before the ast is instantiated. But in terms of behavior, the code block has the same behavior as before.

Reply:
I guess you will revisit this now that you are going to insert the name of the class during instantiation?

Reply:
Thank you for your reply, @mauricioaniche! I am revisiting this point, and I will push the changes as soon as possible.

Reply:
Hi, @mauricioaniche! I have already submitted a new commit with the correction that you had asked. Now, the name of the classes are being inserted during instantiation.